rfcs/0004-fulltext-search.md

```diff
@@ -0,0 +1,275 @@
+    can be easily defined by a subgraph developer ultimately providing 
+    the tools necesssary for front end developers to create effective 
+    user interfaces without the need to middleware.
+     
+## Urgency
+
+A delay in adding the fulltext search feature  will not create issues with 
+current deployments. However, it will represent a realization of part of 
+the long term vision for the query network. Also several high profile users 
+have communicated that it may be a conversion blocker, so implementation 
+should be prioritized. 
+
+## Terminology
+
+- _lexeme_: a basic lexical unit of a language, consisting of one word or 
+  several words, considered as an abstract unit, and applied to a family 
+  of words related by form or meaning.
+- _morphology (linguistics)_: the study of words, how they are formed, 
+  and their relationship to other words in the same language. 
+- _fulltext search index_: the result of lexical and morphological 
+  analysis (stemming) of a set of text documents.  It provides frequency 
+  and location for the language-specific stems found in the text documents 
+  being indexed. 
+- _ranking algorithm_: "Ranking attempts to measure how relevant documents 
+  are to a particular query, so that when there are many matches the most 
+  relevant ones can be shown first." [- Postgres Documentation](https://www.postgresql.org/docs/11/textsearch-controls.html#TEXTSEARCH-RANKING-SEARCH-RESULTS)
+  
+  **Algorithms**:
+  - _standard ranking_: ranking based on the number of matching lexemes.     
+  - _cover density ranking_: Cover density is similar to the standard 
+    fulltext search ranking except that the proximity of matching lexemes 
+    to each other is taken into consideration. This function requires 
+    lexeme positional information to perform its calculation, so it ignores 
+    any "stripped" lexemes in the index.            
+
+## Detailed Design
+
+### Subgraph Definition
+
+Part of the power of fulltext search indexes are their flexibility, so 
+it is important to expose a simple interface to facilitate useful 
+applications of the index and ulimately aim to reduce the need to create 
+new subgraphs for the express purpose of updating fulltext search fields 
+for end-use cases. 
+
+For each fulltext search index a subgraph developer must be able to specify:
+    1. a language, 
+    2. a set of text document fields to include,
+    3. relative weighting for each field in the index.
+    4. a choice of ranking algorithm for sorting query result items
+
+The proposed process of adding one or more fulltext search index to an Entity 
+involves adding one or more fulltext directive to the `_Schema_` type in the 
+subgraph's GraphQL schema.  Each fulltext definition will have four required
+top level parameters: `name`, `language`, `algorithm`, and `include`. 
+The fulltext search definitions will be used to generate queries on the GraphQL
```

Comment:
It will not `generate queries`, it'll generate query fields in the GraphQL schema.